Issue 1375: Add veux plugin docs (#1397)
clay/clay-kiln: docs/veux_plugin.md

---
id: vuex_plugin
title: Vuex Plugins
sidebar_label: Vuex Plugins
---

> When you're working with plugins is slightly tricky have access to the veu object on its context because you don't have direct access to it. The best way to get around this limitation is to dynamically register a store in a created hook on your root component. It is also necessary to namespace your store due to the module nature of Kiln plugins. For more info on `Vuex modules` see official [documentation](https://vuex.vuejs.org/guide/modules.html)

Reviewer: ```suggestion
When you're working with plugins is slightly tricky to get access to the vue object on its context because you don't have direct access to it. The best way to get around this limitation is to dynamically register a store in a created hook on your root component. It is also necessary to namespace your store due to the module nature of Kiln plugins. For more info on `Vuex modules` see official [documentation](https://vuex.vuejs.org/guide/modules.html)
```

Author: Done!!

> When you're working with plugins is slightly tricky to get access to the veu object on its context because you don't have direct access to it. The best way to get around this limitation is to dynamically register a store in a created hook on your root component. It is also necessary to namespace your store due to the module nature of Kiln plugins. For more info on `Vuex modules` see official [documentation](https://vuex.vuejs.org/guide/modules.html)

Reviewer 2: ```suggestion
When you're working with plugins is slightly tricky to get access to the vue object and its context because you don't have direct access to it. The best way to get around this limitation is to dynamically register a store in a created hook on your root component. It is also necessary to namespace your store due to the module nature of Kiln plugins. For more info on `Vuex modules` see official [documentation](https://vuex.vuejs.org/guide/modules.html)
```

Author: Done!! Thanks!!

Example:

```js

const store = {
  // important!
  namespaced: true,
  state: {...}
  actions: {...},
  mutations: {...},
  getters: {...}
}

// The main component
Main = {
  data: () => {...},
  components: {...},

  // register the store under MyStoreNamespace in the created callback

>   created() {this.$store.registerModule('MyStoreNamespace', store)}

Reviewer 3: ```suggestion
  created() {
    this.$store.registerModule('MyStoreNamespace', store);
  }
```
